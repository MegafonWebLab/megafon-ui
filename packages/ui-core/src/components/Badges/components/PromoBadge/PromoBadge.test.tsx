import * as React from 'react';
import { render } from '@testing-library/react';
import PromoBadge, { IPromoBadgeProps } from './PromoBadge';

const dataAttrs: IPromoBadgeProps['dataAttrs'] = {
    root: {
        'data-testid': 'root',
    },
};

describe('<PromoBadge />', () => {
    it('should render PromoBadge', () => {
        const { container } = render(<PromoBadge>text</PromoBadge>);

        expect(container).toMatchSnapshot();
    });

    it('should render with className', () => {
        const { getByTestId } = render(
            <PromoBadge className="custom-class" dataAttrs={dataAttrs}>
                text
            </PromoBadge>,
        );

        expect(getByTestId('root')).toHaveClass('custom-class');
    });

    it('should render with dataAttrs', () => {
        const { queryByTestId } = render(<PromoBadge dataAttrs={dataAttrs}>text</PromoBadge>);

        expect(queryByTestId('root')).toBeTruthy();
    });

    it('should render with type different from default', () => {
        const { queryByTestId } = render(
            <PromoBadge dataAttrs={dataAttrs} type="vip">
                text
            </PromoBadge>,
        );

        expect(queryByTestId('root')).toHaveClass('mfui-promo-badge_type_vip');
    });
});
