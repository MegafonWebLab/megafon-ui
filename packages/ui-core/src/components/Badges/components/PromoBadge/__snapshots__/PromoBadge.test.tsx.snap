// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromoBadge /> should render PromoBadge 1`] = `
<div>
  <div
    class="mfui-promo-badge mfui-promo-badge_type_hit"
  >
    <span
      class="mfui-promo-badge__text"
    >
      text
    </span>
  </div>
</div>
`;
